[로그인 페이지 구현] 클라이언트 Part

akswnd98 opened the pull request:
> #18

**PR 설명**
로그인 기능을 완성하였습니다.
https://giveusmoney.github.io/FanFixiv-user-page/ 에서 테스트 하실 수 있습니다.

**개발된 기능**
- 로그인과 관련된 navigation ui 이슈 제거
- 로그인 기능 (twitter 제외)

**레퍼런스**
- 현재 dev server에 올려둔 상태입니다. https://giveusmoney.github.io/FanFixiv-user-page/

akswnd98 committed:
feat: 로그인 기능

- recoil setting: 전역상태관리 라이브러리를 세팅하였습니다.
- 로그인, 로그아웃 화면 및 로그인, 로그아웃 기능 (아직 twitter 로그인은 안됨)
- Drawer와 Navigation에 z-index 지정 -> 안가려지게 수정
- Avatar의 Chevron icon이 클릭과 호버가 되게 수정
- Drawer의 소스 위치를 Avatar 밑 -> Navigation 밑으로
- Drawer Item 안 뭉게지게 수정
- react dotenv 도입

akswnd98 committed:
feat: navigation 이슈 반영

- 로그인시 Search 박스가 움직이는 현상 제거
- 19금 표시가 로그인 이후에도 보이는 현상 제거

akswnd98 committed:
Merge branch 'develop' into f/bgstotem/#18

akswnd98 committed:
feat: .env에 REACT_APP_PUBLIC_URL 추가

배포 시에 REACT_APP_PUBLIC_URL이 달라질 수 있으므로 (예: gh-pages에선 /FanFixiv-user-page) .env에 추가하였습니다.

akswnd98 committed:
fix: Drawer위치 Avatar 밑으로 고정

Navigation width를 가변 지정하고 나서 Drawer가 Avatar과 떨어지는 문제가 발생하여 수정했습니다

akswnd98 committed:
fix: axios의 헤더 설정이 gh-pages에서 잘 안되는 현상 제거

initializeAxios()의 호출 타임이 App의 렌더링 이후로 되어 있어서
두번 렌더링하는 local (React.StrictMode)의 경우 header에 값이 대입이 되어있어
정상 동작하지만 gh-pages에서 구동하는 경우 정상 로그인이 안되는 문제가 있었습니다.

dalcon10028 committed:
chore: 디렉터리 구조 정리, 플러그인 추가

dalcon10028 committed:
feat: 회원가입 페이지 구현

dalcon10028 committed:
docs: env gitignore 추가, env 가이드 작성

dalcon10028 committed:
chore: settings.json 추가

dalcon10028 committed:
Merge branch 'develop' into f/dalcon10028/#20

dalcon10028 commented on src/components/BlockMessage/index.tsx:
```diff
@@ -0,0 +1,42 @@
+
+export type PropsType = {
+  message: string;
+};
```
이거 이전에는 interface 파일 분리하기로 했던 것 같아서 이거 이따가 다같이 얘기 해요

dalcon10028 commented on src/components/Navigation/Buttons/index.tsx:
```diff
@@ -29,13 +32,37 @@ const MarginWrapper = styled.div`
+    };
+    // checkLogin();
+    validateLogin();
+  }, []);
```
이거는 axios설정 제가 따로 빼놔서 이것도 이따가 같이 이야기 해보아요

dalcon10028 commented on src/index.tsx:
```diff
@@ -1,13 +1,38 @@
+  initializeAxios();
+  const root = ReactDOM.createRoot(
+    document.getElementById('root') as HTMLElement,
+  );
```
요거 위치도 같이 고민해 보아요

dalcon10028 commented on src/states/avatarDrawer/index.ts:
```diff
@@ -0,0 +1,8 @@
+import { atom } from 'recoil';
+
+const avatarDrawer = atom<boolean>({
+  key: 'avatarDrawerState',
+  default: false,
+});
```
요거는 props 를 안쓰고 분리하신 이유가 몬가여?

dalcon10028 reviewed (changes requested):
고생하셨습니다 이따 몇가지 의논할 게 있어서 조정 후 반영 부탁드립니다 :)

akswnd98 committed:
Merge branch 'f/dalcon10028/#20' into f/bgstotem/#18

akswnd98 committed:
fix: 로그인관련 api 추가

- 로그인 관련 userApi를 추가하고 코드를 userApi를 사용하도록 수정하였습니다.
- 부수적으로 root의 axios init 코드와 refresh 코드가 userApi의 axiosInstance 초기화 쪽으로 이동되었습니다.

akswnd98 committed:
fix: userApi.getProfile 에러 제거

- getProfile에서 exam5432@naver.com을 특정에서 요청하는 버그
- get이 아닌 post로 요청하는 버그

akswnd98 committed:
fix: 로그인 팝업 비밀번호 찾기, 회원가입 밑줄

호버시 밑줄이 표시 되며 색이 검정색으로 변합니다.

akswnd98 committed:
fix: 로그인 버튼들 hover시 색상 변경

로그인 버튼의 평소와 hover시의 색상을 반대로 바꿨습니다.
twitter 로그인 버튼의 hover 색상을 추가했습니다.

akswnd98 committed:
feat: userApi header setter 추가

userApi에서 header setting하는 메소드 추가 (setDefaultCommonHeader)

Roharui commented on src/pages/twitter/login/index.tsx:
```diff
@@ -0,0 +1,13 @@
+    if (token) {
+      window.localStorage.setItem('access-token', token);
+    }
+    location.href = '/';
```
이 부분을 `REACT_APP_PUBLIC_URL`로 바꿔야 할듯합니다.

Roharui commented on src/components/layout/Navigation/Drawer/Items/Empty/index.tsx:
```diff
@@ -1,6 +1,6 @@
 import styled from '@emotion/styled';
 
-import theme from '../../../../../../../assets/theme/theme';
+import theme from '../../../../../../assets/theme/theme';
```
이 부분 `@src`로 고칠수는 없나요?

akswnd98 replied:
6중은 좀 심했네요.
@assets로 바꾸겠습니다.

Roharui commented on src/apis/userApi.ts:
```diff
@@ -0,0 +1,75 @@
+    return this.post<LoginResponseDto, LoginRequestDto>(
+      `${this.authResource}/login`,
+      {
+        id: email,
```
`id`라고 되어있는데 백엔드에서 `email`로 고칠수 있습니다. 헷갈릴 여지가 있다면 바꿔 드리겠습니다.

akswnd98 replied:
email만 있으니까 email로 갔으면 좋겠습니다.

akswnd98 committed:
feat: twitter 로그인 기능 추가

twitter로그인 버튼을 누르면 oauth2 로그인을 시작합니다.

Roharui reviewed (approved):
로그인 이후 19금 Switch 표시와 Drawer 그림자 추가되었나요?

akswnd98 committed:
fix: 19금 Switch가 표시되는 타이밍 변경

비 로그인 시 -> 로그인 시로 변경

akswnd98 committed:
Merge pull request #33 from GiveUsMoney/f/bgstotem/#32

fix: 19금 Switch가 표시되는 타이밍 변경

akswnd98 committed:
fix: code form 개선

- ../../../ -> @로
- 사용되지 않은 참조 삭제

akswnd98 committed:
fix: datastruct code 형식 변경

이전에 회의 결과로 export type xxx 대신 interface xxx 로 바꾸고 datastruct 코드는 따로 모아 interface.ts에
정리하기로 했습니다.

akswnd98 committed:
feat: logout 시 logout api호출

logout시 서버에 logout 시그널을 보냅니다.

akswnd98 committed:
fix: login popup 코드 개선

Component에서 logic 분리

akswnd98 committed:
fix: logout 코드 개선

Component에서 logic 분리

akswnd98 committed:
fix: Navigation Buttons 코드 개선

Component에서 logic 분리

akswnd98 committed:
fix: login 관련 global interface 추가

login 상태를 모델을 통해서 관리 할 수 있게 추가했습니다.

custom hook을 알고나서 react가 다르게 보입니다.

akswnd98 committed:
Merge branch 'develop' into f/bgstotem/#18
